options/options_helper.cc

```diff
@@ -969,6 +969,20 @@ Status OptionTypeInfo::Parse(const ConfigOptions& config_options,
       ConfigOptions copy = config_options;
       copy.invoke_prepare_options = false;
       return parse_func_(copy, opt_name, opt_value, opt_addr);
+    } else if (opt_name == "read_amp_bytes_per_bit") {
+      // A temporary hack to work around a bug in 6.10, 6.11, 6.12, 6.13 and
+      // 6.14. The bug will write out 8 bytes to OPTIONS file from the starting
+      // address of BlockBasedTableOptions.read_amp_bytes_per_bit which is
+      // actually a uint32. Consequently, the value of read_amp_bytes_per_bit
+      // written in the OPTIONS file is wrong. From 6.15, RocksDB will
+      // try to parse the read_amp_bytes_per_bit from OPTIONS file as a uint32.
+      // To be able to load OPTIONS file generated by affected releases before
+      // the fix, we need to manually parse read_amp_bytes_per_bit with this
+      // special hack.
+      assert(type_ == OptionType::kUInt32T);
+      uint64_t read_amp_bytes_per_bit = ParseUint64(opt_value);
+      EncodeFixed32(opt_addr, static_cast<uint32_t>(read_amp_bytes_per_bit));
```

Comment:
I too am confused as to the need for the EncodeFixed32.  In my experiment here, the following worked as well:

 *addr = static_cast<uint32_t>(ParseUint64(value));

Reply:
The `addr` or `opt_addr` is of type `char*`, to correctly store an integer at this address, we need to consider endianness.

Reply:
No, this code is wrong. If you were correct, the entirety of ParseOptionHelper would be wrong. A function like ParseUInt64 returns a value in native endianness. It needs to be stored in the struct in native endianness. EncodeFixed32 stores it in the struct in little endian, including on big endian machines.